lib/astarte_data_access/c_system.ex

```diff
@@ -0,0 +1,126 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 - 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule CSystem do
```

Comment:
Libraries should at least namespace their modules, so as not to pollute the global environment. See https://hexdocs.pm/elixir/1.12.3/library-guidelines.html#avoid-defining-modules-that-are-not-in-your-namespace
```suggestion
defmodule Astarte.DataAccess.CSystem do
```